sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,331 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.Protobuf.Ucfg;
+using SonarAnalyzer.SymbolicExecution.ControlFlowGraph;
+using UcfgLocation = SonarAnalyzer.Protobuf.Ucfg.Location;
+
+namespace SonarAnalyzer.Security.Ucfg
+{
+    public class UniversalControlFlowGraphBuilder
+    {
+        public static readonly string Const = "\"\""; // Same as Java implementation
+        private readonly BlockIdMap blockId = new BlockIdMap();
+        private readonly SemanticModel semanticModel;
+        private readonly InstructionBuilder instructionBuilder;
+        private readonly IControlFlowGraph cfg;
+
+        public UniversalControlFlowGraphBuilder(SemanticModel semanticModel, IControlFlowGraph cfg)
+        {
+            this.semanticModel = semanticModel;
```

Comment:
Argument checks/assertions that not null.

Reply:
Adding null check + a "pointless" test to keep up with the coverage does not seem very reasonable imho (and also there are just a few argument null checks in this project)

Reply:
`SemanticModel` cannot be null when it is taken from the analysis context (or at least we never did null checks and we never fail for such reason). The `semanticModel` argument will be `null` if you explicitly pass `null` as parameter... Possible, but unlikely in my opinion.